internal/util/go_with_wait_group_test.go

```diff
@@ -0,0 +1,36 @@
+package util_test
+
+import (
+	"github.com/redhat-et/apex/internal/util"
+	"github.com/stretchr/testify/assert"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+func TestGoWithWaitGroup(t *testing.T) {
+
+	counter1 := atomic.Int32{}
+	for i := 0; i < 10; i++ {
+		// run a few functions in parallel
+		util.GoWithWaitGroup(nil, func() {
+			time.Sleep(200 * time.Millisecond)
+			counter1.Add(1)
+		})
+	}
+	assert.Equal(t, int32(0), counter1.Load())
```

Comment:
Is there some way we can know this would never fail? It seems like it could fail if for some reason one of the goroutines was scheduled after 200 milliseconds and before getting to this assertion. I know it seems unlikely, but who knows, maybe under really heavy load?

A couple of alternatives:

- Just rely on the test not crashing on a nil waitgroup.

- Right now there's no check that verifies that the goroutine ran. You could drop the sleep inside the routines and do a different type of assertion.  In psuedo-code:

```
for {
    break if the counter has reached 10
    assert a test failure if we've waited too long (10 seconds or something absurdly long)
    otherwise sleep a bit and check again
}
```

Reply:
good idea.. I just took a crack it doing all of that.  should be better.